### DISCUSSION TITLE
Can't install STT

### DISCUSSION THREAD

**Discussion by Emekaborisama** [coqui-ai/STT#1847] [Q&A]
I tried installing STT with pip but I keep getting this error below

`ERROR: Could not find a version that satisfies the requirement stt`
`ERROR: No matching distribution found for stt`

**Comment by reuben** [comment]
Can you provide information about your system? What operating system, architecture, version of Python, version of pip, etc. And just in case, try updating pip before you install STT: `python -m pip install -U pip`

**Reply by Emekaborisama** [reply]
i use macbook pro 2015, python version 3.6.2, pip 3.6. i have updated the Pip and it still return the same error

**Reply by reuben** [reply]
What's the output of `python -m pip --version` after you updated it? And does it still error if you install with `python -m pip install stt`?

**Reply by reuben** [reply]
Also, what's the version of your operating system?

**Reply by Emekaborisama** [reply]
The output of the `pip -m pip --version` is `pip 21.0.1 from /Users/emekaborisama/Downloads/New_data_file/couqui/venv-stt/lib/python3.6/site-packages/pip (python 3.6)`

Yes, after updating it, I still get the same error.

**Reply by Emekaborisama** [reply]
High Serria

**Reply by reuben** [reply]
Ah, OK. macOS High Sierra is unsupported by Apple so I would recommend upgrading. Our packages are compatible with macOS 10.14 Mojave and newer, but you can probably get away with simply downloading and renaming the package file as a hack (not recommended):

```bash
curl -LO https://files.pythonhosted.org/packages/e1/fc/e06e0829d26ca2c812d6d6860dc999ffa296534012dcaca077289ea7859c/STT-0.10.0a4-cp36-cp36m-macosx_10_14_x86_64.whl
mv STT-0.10.0a4-cp36-cp36m-macosx_10_14_x86_64.whl STT-0.10.0a4-cp36-cp36m-macosx_10_13_x86_64.whl
python -m pip install STT-0.10.0a4-cp36-cp36m-macosx_10_13_x86_64.whl
```

**EDIT**: This should no longer be needed as of version 0.10.0a5. Just do a normal `python -m pip install stt`.

**Reply by Emekaborisama** [reply]
it installed well but when i run  `stt --model stt-0.9.3-models.pbmm --scorer stt-0.9.3-models.scorer --audio sample.wav`

I get an error `RuntimeError: CreateModel failed with 'Failed to initialize memory mapped model.' (0x3000)`

**Reply by Emekaborisama** [reply]
Dont worry i didnt use the right dir name for the scorer

**Reply by Emekaborisama** [reply]
Here is the error i am getting now

 `Loading scorer from files coqui-stt-0.9.3-models.scorer
libc++abi.dylib: terminating with uncaught exception of type util::EndOfFileException: native_client/kenlm/util/file.cc:264 in void util::ErsatzPRead(int, void *, std::size_t, uint64_t) threw EndOfFileException because `ret == 0'.
End of file for reading 2 bytes at 12007440 from fd 4`

**Reply by reuben** [reply]
Looks like you have a corrupted scorer file somehow.

**Reply by reuben** [reply]
FWIW I've just uploaded the new 0.10.0a5 version which should be compatible with older versions of macOS with a simple `python -m pip install stt` instead of the hack I described above.

**Reply by jeffreyrlewis** [reply]
Unfortunately, I'm back to the original problem: ERROR: No matching distribution found for stt

(macOS Monterey, python3.9.10, pip pip 22.0.3)

Thoughts?

**Comment by FlorianO** [comment]
Hi, 
i have probably the same problem.

- macOS Monterey 12.2
- Python 3.9.12
- pip 22.0.4

I follow the Quickstart from the documentation:
`python -m pip install coqui-stt-model-manager`

```ERROR: Cannot install coqui-stt-model-manager==0.0.10, coqui-stt-model-manager==0.0.11, coqui-stt-model-manager==0.0.12, coqui-stt-model-manager==0.0.14, coqui-stt-model-manager==0.0.15, coqui-stt-model-manager==0.0.16, coqui-stt-model-manager==0.0.17, coqui-stt-model-manager==0.0.18, coqui-stt-model-manager==0.0.19, coqui-stt-model-manager==0.0.20, coqui-stt-model-manager==0.0.8 and coqui-stt-model-manager==0.0.9 because these package versions have conflicting dependencies.

The conflict is caused by:
    coqui-stt-model-manager 0.0.20 depends on stt==1.3.0
    coqui-stt-model-manager 0.0.19 depends on stt==1.1.0
    coqui-stt-model-manager 0.0.18 depends on stt==1.0.0
    coqui-stt-model-manager 0.0.17 depends on stt==0.10.0a13
    coqui-stt-model-manager 0.0.16 depends on stt==0.10.0a13
    coqui-stt-model-manager 0.0.15 depends on stt==0.10.0a13
    coqui-stt-model-manager 0.0.14 depends on stt==0.10.0a13
    coqui-stt-model-manager 0.0.12 depends on stt==0.10.0a6
    coqui-stt-model-manager 0.0.11 depends on stt==0.10.0a6
    coqui-stt-model-manager 0.0.10 depends on stt==0.10.0a6
    coqui-stt-model-manager 0.0.9 depends on stt==0.10.0a6
    coqui-stt-model-manager 0.0.8 depends on stt==0.10.0a6

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts```

**Reply by FlorianO** [reply]
I can't install stt manually.
```
pip install stt
ERROR: Could not find a version that satisfies the requirement stt (from versions: none)
ERROR: No matching distribution found for stt
```

**Reply by wasertech** [reply]
> `ERROR: Could not find a version that satisfies the requirement stt (from versions: none)`

If you see this, it means there is no wheels of `stt` available for your python version and CPU architecture combination.

If no distribution was found, you have the following options:
1. You can help Coqui to build wheel for your platform with the help of CI (checkout #2023 ).
2. You can build stt and make the python bindings yourself ([docs](https://stt.readthedocs.io/en/latest/BUILDING.html#building-binaries)).

**Comment by Vrajs16** [comment]
I can't install stt with pip either. tts works though?

**Reply by FlorianO** [reply]
At least it tries to. 
But it fails to install numpy==1.19.5
`clang: error: the clang compiler does not support 'faltivec', please use -maltivec and include altivec.h explicitly`
I already have a newer version of numpy installed (1.22.3)

**Reply by FlorianO** [reply]
`pip index versions stt`
`WARNING: pip index is currently an experimental command. It may be removed/changed in a future release without prior warning.`
`ERROR: No matching distribution found for stt`

If i do this for tts it shows:
`pip index versions tts`
`WARNING: pip index is currently an experimental command. It may be removed/changed in a future release without prior warning.`
`tts (0.6.2)`
`Available versions: 0.6.2, 0.6.1, 0.6.0, 0.5.0, 0.4.2, 0.4.1, 0.4.0, 0.3.1, 0.3.0, 0.2.2, 0.2.1, 0.2.0, 0.1.3, 0.1.2, 0.1.1, 0.1.0`

**Reply by Vrajs16** [reply]
Hey Florian0. What version of Python are you running. If you are running a m1 mac computer, then you have to either upgrade your python version to the latest version to support the m1 mac or compile from source.

**Comment by Loqova** [comment]
Same issue with stt. https://github.com/abhirooptalasila/AutoSub/issues/65

**Comment by strarsis** [comment]
Updating Ubuntu 20.x LTS to 22.x LTS fixed the issue (Ubuntu on WSL 2). 
The newer `stt` package releases apparently require a newer Ubuntu distribution.

Edit: `coqui-stt-model-manager` can't be installed, as it depends on an older `stt` `1.0.0`.
